usr/sbin/p11sak/p11sak.c

```diff
@@ -1356,6 +1368,7 @@ static CK_RV tok_key_get_key_type(CK_SESSION_HANDLE session, CK_OBJECT_HANDLE hk
                 ret);
         return ret;
```

Comment:
I think buffer also needs to be free'd here (probably at all error exits). Think about providing an error exit label for this function.

Reply:
Fixed. buffer and label need to be freed in the while(1) loop.

Reply:
buffer still needs to be freed on below switch statement's error return.